core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -183,6 +185,11 @@ class KafkaApis(val requestChannel: RequestChannel,
     val updateMetadataResponse =
       if (authorize(request.session, ClusterAction, Resource.ClusterResource)) {
         replicaManager.maybeUpdateMetadataCache(correlationId, updateMetadataRequest, metadataCache)
+        if (controller.isActive && !updateMetadataRequest.partitionStates.isEmpty) {
```

Comment:
Good point. I think it should be okay to remove that check.

In what scenarios could a controller failover and the broker is still healthy enough to respond?